### ISSUE TITLE
Merging repository boards allow to view a single burndown for both repos

### ISSUE THREAD

**Issue by idonahue** [ZenHubIO/support#247]

**Comment by aupright** [comment]
Thanks for the feedback here @idonahue!

There have been a few similar requests here on our support repo for this feature—we've issued it up internally and are currently exploring a few different approaches. I'll keep you updated on our progress here and reach out when we have an ETA! :beers:

**Comment by meggiebot** [comment]
As a project manager, I would like to see a burndown chart that spans multiple repositories.

The burndown could either 1) show the same repositories that are merged into the board or 2) have its own repositories/merged repositories.

Repositories would need the same milestone in them for issues to show in the burndown chart.

This is important to track releases/milestones that span multiple teams or repositories.

As our team was brainstorming around this, we though of a few ways to work around milestones that span multiple repositories.

First, ZenHub could create their own concept of milestones instead of using GitHub's milestones. By doing that ZenHub could also then add the ability to have multiple milestones assigned to an issue (ie sprint milestone, release milestone) so that burndowns could be tracked at multiple levels.

Second, ZenHub could still use GitHub's milestones and just aggregate them based on name. Users would have to make sure the milestone in the different repos was exactly the same.

**Comment by illfang** [comment]
As a lead developer I would like to see a summarized progress for a milestone which includes changes in several repositories. Using the Board I can already track the progress of the different tasks but there is no way to quickly confirm the progress using a Burndown chart in this scenario.

**Comment by ivanmladek** [comment]
+1

**Comment by SeriouslyAwesome** [comment]
> Second, ZenHub could still use GitHub's milestones and just aggregate them based on name. Users would have to make sure the milestone in the different repos was exactly the same.

Until something like this is implemented, I'm looking to using GitHub's API to create milestones with the same attributes on selected repos from a single form.

**Comment by devinivy** [comment]
Given that milestones with the same name in different repos are merged in the board view, it would make sense for the burndown charts to work similarly.

**Comment by arifshanji** [comment]
+1

**Comment by chadfurman** [comment]
as the only developer on a project, I would like to have an issue in both a weekly Sprint milestone as well as an over-all feature milestone.  For example, I would like be able to see a burn-down chart for the week to track my velocity while still seeing a burn-down chart for the over-all feature that this ticket is a part of (i.e. issue: User account profile page needs to display Milestone: User Profiles, Sprint 1,  Issue:  user account profile needs to allow changing of notification settings, etc. Milestone: User Profiles, Sprint 2)

**Comment by tjhanley** [comment]
+1

**Comment by azenMatt** [comment]
Hi all, 

Just a quick update here to thank you for your patience while this issue is worked on! 

I have added the confirmed label as this is in our roadmap for Q1, so we should have this functionality with you shortly!

**Comment by wesoudshoorn** [comment]
Great @azenMatt, we're really waiting on this :)

**Comment by chadfurman** [comment]
Thank you, @azenMatt!  I just looked at a ticket in my backlog and was like, "I'd really like to put that in two milestones..."

**Comment by FrankSchuil** [comment]
One up.

**Comment by jonnyAnd** [comment]
+1

**Comment by tiagotaveiragomes** [comment]
+1

**Comment by azenMatt** [comment]
This is now live in v2.15.61 :boom:

![image](https://cloud.githubusercontent.com/assets/7584089/13515989/ca4d1658-e16a-11e5-8537-0a16c9f8d01b.png)

Based on repositories connected together in a ZenHub board, if a milestone has the same name and end date, it will appear together on the Burndown chart. This is the same logic used to view the same milestone on a multi-repository board. 

Using the check boxes at the top right, you can select the burndown for a different combination of repos that it is considered matching to. 


We will have a releasing coming down the pipeline that will make managing the multi repository milestones much easier so that you don't need to individual create them exactly the same for each repo. 

Any feedback, please let me know!

**Comment by FrankSchuil** [comment]
Is v2.15.61 live live? I refreshed my page and tried to use the functionality, but the checkboxes don't show up as in the screenshot. I do have merged milestones like in the board that work as expected.

**Comment by meggiebot** [comment]
What is the status of choosing something other than a milestone for the burndown chart?

**Comment by chadfurman** [comment]
if I can't have two milestones on a ticket, I have to take a ticket out of the over-all milestone for the project and assign it to a sprint to track burndown progress.

I'm with @meggiebot -- if we can only have one milestone per issue, then something other than milestones needs to produce burndowns.  I want to see the progress in a sprint, as well as the progress towards over-all project milestones.

**Comment by idonahue** [comment]
Thanks guys! You're awesome :-)

**Comment by meggiebot** [comment]
Yep, I'm on the same page as @chadfurman. We use milestones for sprints so it's not  super interesting to see the sprint burn-down--even if it is across multiple repos. Though this is a great first step.

**Comment by azenMatt** [comment]
For the non-milestone burndowns - the start of Q2 is looking good right now to start seeing the next stage of these reports take shape. I will keep #431 up to date with our progress here. 

@chadfurman - two milestone tickets are currently not on our roadmap, but we do have an upcoming release around Epics that should help solve this for you (in terms of how you are currently having to use a milestone for both a project and a sprint). As this will give an additional grouping type (which we will also allow reporting over)

**Comment by jkauszler** [comment]
Has there been any progress on this release: "We will have a releasing coming down the pipeline that will make managing the multi repository milestones much easier so that you don't need to individual create them exactly the same for each repo." ?

I haven't seen how to do this, though I many not be looking in the correct place. 

Thanks!

**Comment by azenMatt** [comment]
> Has there been any progress on this release: "We will have a releasing coming down the pipeline that will make managing the multi repository milestones much easier so that you don't need to individual create them exactly the same for each repo." ?


Not yet, we are still struggling with the UX for how this would work without being really confusing. 

This is still a 'confirmed' item though in that we very much want to get this one out! 

The best place to follow along is #597 👍
